contracts/Pool.sol

```diff
@@ -1,8 +1,35 @@
 // SPDX-License-Identifier: GPL-3.0-only
 pragma solidity ^0.6.2;
 
-/// @notice Funding pool contract.
-/// Allows sending ETH automatically to a configurable set of receivers.
+/// @notice Funding pool contract. Automatically sends funds to a configurable set of receivers.
+///
+/// The contract has 2 types of users: the senders and the receivers.
+///
+/// A sender has some funds and a set of addresses of receivers, to whom he wants to send funds.
+/// In order to send there are 3 conditions, which must be fulfilled:
+///
+/// 1. There must be funds on his account in this contract.
+///    They can be added with `topUp` and removed with `withdraw`.
+/// 2. Total amount sent to the receivers on each block must be set.
+///    This is done with `setAmountPerBlock`.
+/// 3. A set of receivers must be non-empty.
+///    Receivers can be added, removed and updated with `setReceiver`.
+///    Each receiver has a weight, which is used to calculate how the total sent amount is split.
+///
+/// Each of these functions can be called in any order and at any time, they have immediate effects.
+/// When all of these conditions are fulfilled, on each block the configured amount is being sent.
+/// It's extracted from the `withdraw`able balance and transferred to the receivers.
+/// The process continues automatically until the sender's balance is empty.
+///
+/// The receiver has an account, from which he can `collect` funds sent by the senders.
+/// The available amount is updated every `cycleBlocks` blocks,
```

Comment:
Same here, the amount isn't updated every `cycleBlocks`, but rather, the amount is calculated based on `cycleBlocks`, at the time of collection.